### PR TITLE
Hydra test coverage for column table update and delete.

### PR DESCRIPTION
**Pull request by sonalsagarwal**
## Changes proposed in this pull request
Create a new set of tests in dmlOps.bt with :
- Column tables tests for update and delete testing using JDBC client, validation against derby after each operation. Full table validation against derby at the end of the test.
- Column tables tests for update and delete testing using snappyJob, validation against derby after each operation. Full table validation against derby at the end of the test.
- Column tables tests for update and delete testing in smart connector mode, validation against derby after each operation. Full table validation against derby at the end of the test.
- Lead and server HA tests for above connection types with validation
- API coverage for update, delete and insert. Tests changes yet to complete.

## Patch testing
Ran the new tests with row tables. Will also run on column tables.

## ReleaseNotes.txt changes
N/A
## Other PRs 
N/A

### CODE REVIEW

File: dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java [hunk 1]
```diff
@@ -1991,14 +2000,18 @@ protected void executeSparkJob(Vector jobClassNames, String logFileName) {
...
+        }
         command = snappyJobScript + " --class " + userJob +
             " --master spark://" + masterHost + ":" + masterPort + " " +
             SnappyPrms.getExecutorMemory() + " " +
             SnappyPrms.getSparkSubmitExtraPrms() + " " +
             " --conf spark.executor.extraJavaOptions=-XX:+HeapDumpOnOutOfMemoryError" +
             " --conf spark.extraListeners=io.snappydata.hydra.SnappyCustomSparkListener" +
             " " + snappyTest.getUserAppJarLocation(userAppJar, jarPath) + " " +
-            SnappyPrms.getUserAppArgs() + " " + primaryLocatorHost + ":" + primaryLocatorPort;
+            userAppJars + " " + primaryLocatorHost + ":" + primaryLocatorPort;
```
Comment (swatisawant) [inline dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java]: Change name to userAppArgs.

File: dtests/src/test/java/io/snappydata/hydra/testDMLOps/SnappyDMLOpsUtil.java [hunk 2]
```diff
@@ -0,0 +1,1049 @@
...
+  public static boolean isHATest = false;
+  protected static hydra.blackboard.SharedLock dmlLock;
+
+  protected static SnappyDMLOpsUtil testInstance;
+  public static DerbyTestUtils derbyTestUtils;
+
+  public static void HydraTask_initialize() {
+    if (testInstance == null)
+      testInstance = new SnappyDMLOpsUtil();
+    hasDerbyServer = TestConfig.tab().booleanAt(Prms.manageDerbyServer, false);
```
Comment (swatisawant) [inline dtests/src/test/java/io/snappydata/hydra/testDMLOps/SnappyDMLOpsUtil.java]: Add this initialization at variable creation itself.

File: dtests/src/test/java/io/snappydata/hydra/testDMLOps/SnappyDMLOpsUtil.java [hunk 3]
```diff
@@ -0,0 +1,1049 @@
...
+      } else if (dmlOp.equals(DELETE.getOpType())) {
+        return DELETE;
+      } else return null;
+    }
+  }
+
+  //ENUM for Connection Type
+  public enum ConnType {
+    JDBC("jdbc"),
+    SNAPPY("snappy"),
```
Comment (swatisawant) [inline dtests/src/test/java/io/snappydata/hydra/testDMLOps/SnappyDMLOpsUtil.java]: Rename it to Embedded.

File: dtests/src/test/java/io/snappydata/hydra/testDMLOps/SnappyDMLOpsUtil.java [hunk 4]
```diff
@@ -0,0 +1,1049 @@
...
+    else
+      selectThreads = new ArrayList<>();
+    if (!selectThreads.contains(testInstance.getMyTid())) {
+      selectThreads.add(testInstance.getMyTid());
+      SnapshotIsolationDMLOpsBB.getBB().getSharedMap().put("selectThreads", selectThreads);
+    }
+    testInstance.releaseDmlLock();
+  }
+
+  public static void HydraTask_initializeTablesMetaData() {
```
Comment (swatisawant) [inline dtests/src/test/java/io/snappydata/hydra/testDMLOps/SnappyDMLOpsUtil.java]: Change name of the method to saveTableMetaData

File: dtests/src/test/java/io/snappydata/hydra/testDMLOps/SnappyDMLOpsUtil.java [hunk 5]
```diff
@@ -0,0 +1,1049 @@
...
+      case INSERT:
+        Log.getLogWriter().info("Performing insert operation.");
+        performInsert();
+        break;
+      case UPDATE:
+        Log.getLogWriter().info("Performing update operation.");
+        if (connType.equals(ConnType.JDBC))
+          performUpdate();
+        else
+          performUpdateInSnappy(connType);
```
Comment (swatisawant) [inline dtests/src/test/java/io/snappydata/hydra/testDMLOps/SnappyDMLOpsUtil.java]: Rename method names. Seems quite confusing.  e.g. performUpdate() to performUpdateUsing JDBC()
and performUpdateInSnappy() to performUpdateUsingSNC()

File: dtests/src/test/scala/io/snappydata/hydra/testDMLOps/ValidateDMLOpApp.scala [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+ */
+
+package io.snappydata.hydra.testDMLOps
+
+import java.io.{File, FileOutputStream, PrintWriter}
+
+import org.apache.spark.sql.{SQLContext, SnappyContext}
+import org.apache.spark.{SparkConf, SparkContext}
+
+object ValidateDMLOpApp {
```
Comment (swatisawant) [inline dtests/src/test/scala/io/snappydata/hydra/testDMLOps/ValidateDMLOpApp.scala]: Change job class Name.

File: dtests/src/test/scala/io/snappydata/hydra/testDMLOps/ValidateDMLOpJob.scala [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+import java.io.{File, FileOutputStream, PrintWriter}
+
+import scala.util.{Failure, Success, Try}
+
+import com.typesafe.config.Config
+
+import org.apache.spark.sql.{SnappyJobValid, SnappyJobValidation, SnappySQLJob, SnappySession}
+
+class ValidateDMLOpJob extends SnappySQLJob {
```
Comment (swatisawant) [inline dtests/src/test/scala/io/snappydata/hydra/testDMLOps/ValidateDMLOpJob.scala]: Change job class Name

File: dtests/src/test/scala/io/snappydata/hydra/testDMLOps/ValidateUpdateOpUsingAPI_Job.scala [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+import scala.util.{Failure, Success, Try}
+
+import com.typesafe.config.Config
+import util.TestException
+import java.util
+
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.{SQLContext, SnappyJobValid, SnappyJobValidation, SnappySQLJob, SnappySession}
+
+class ValidateUpdateOpUsingAPI_Job extends SnappySQLJob {
```
Comment (swatisawant) [inline dtests/src/test/scala/io/snappydata/hydra/testDMLOps/ValidateUpdateOpUsingAPI_Job.scala]: Change the job class name.

**Comment by sonalsagarwal** [comment]
Will include the review comments later.

**Pull request merged**